File: Logistic Regression/octave/costFunction.m
function [J, deltaJ] = costFunction(X, y, theta)
%   Logistic regression
%   Cost and gradient 
    m = length(y);
    
    % HYPOTHESIS
    z = X * theta;
    %FILE: sigmoid.m
    h = sigmoid(z);

    % COST
    a = - y' * log(h);
    b = (1 - y)' * log(1 - h);
    J = 1/m * (a - b);

    % GRADIENT
    deltaJ = 1/m * X' * ( h - y );

end

File: Logistic Regression/python/costFunction.py
import numpy as np
from python.sigmoid import sigmoid

def computeCost(X, y, theta):
#   Logistic regression
#   Cost and gradient 
    m = y.size

#   HYPOTHESIS
    z = X.dot( theta )
    h = sigmoid(z)

    # COST
    a = - y.T.dot( np.log(h) )
    b = (1 - y).T.dot( np.log(1 - h) )
    J = 1/m * (a - b)

    # GRADIENT
    deltaJ = 1/m * X.T.dot( ( h - y ) )
    return J, deltaJ